fixtures/dom/src/components/fixtures/text-inputs/index.js

```diff
@@ -0,0 +1,66 @@
+const React = window.React;
+
+const TextInputFixtures = React.createClass({
+  getInitialState() {
+    return {
+      color: '#ffaaee',
+    };
+  },
+
+  renderControlled(type) {
+    let id = `controlled_${type}`;
+
+    let onChange = e => {
+      let value = e.target.value;
+      if (type === 'number') {
+        value = value === '' ? '' : parseFloat(value, 10) || 0;
+      }
+      this.setState({
+        [type] : value,
+      });
+    };
+
+    let state = this.state[type] || '';
+
+    return (
+      <div key={type} className="field">
+        <label htmlFor={id}>{type}</label>
+        <input id={id} type={type} value={state} onChange={onChange} />
+        &nbsp; &rarr; {JSON.stringify(state)}
+      </div>
+    );
+  },
+
+  renderUncontrolled(type) {
+    let id = `uncontrolled_${type}`;
+    return (
+      <div key={type} className="field">
+        <label htmlFor={id}>{type}</label>
+        <input id={id} type={type} />
+      </div>
+    );
+  },
+
+  render() {
+    // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
+    let types = [
+      'text', 'email', 'number', 'url', 'tel',
+      'color', 'date', 'datetime-local',
```

Comment:
AFAIK  `datetime-local` is still supported as `datetime` in some browsers, I'm not sure if that's worth adding or noting tho.